tests/test_compile_guard.py

```diff
@@ -324,3 +324,39 @@ class Top(m.Circuit):
     m.compile(f"build/{basename}", Top)
     assert m.testing.check_files_equal(
         __file__, f"build/{basename}.v", f"gold/{basename}.v")
+
+
+def test_compile_guard_anon_driven_internal():
+
+    class _Top(m.Circuit):
+        io = m.IO(I=m.In(m.Bit), O=m.Out(m.Bit)) + m.ClockIO()
+
+        with m.compile_guard("COND", defn_name="COND_compile_guard"):
+            x = m.Bit(name="x")
+            x.undriven()  # x is driven internally
+            out = m.Register(m.Bit)()(x)
+
+        io.O @= io.I
+
+    m.compile("build/test_compile_guard_anon_driven_internal", _Top,
+              output="mlir")
+    assert m.testing.check_files_equal(
+        __file__, "build/test_compile_guard_anon_driven_internal.mlir",
+        "gold/test_compile_guard_anon_driven_internal.mlir")
+
+
+def test_compile_guard_anon_driving_external():
```

Comment:
I think should should not be valid magma. Either we should explicitly disallow driving outside values from within a compile guard *or* implicitly convert them to `compile_guard_select`.

(BTW the generated MLIR is not "valid" in that it contains undefined SSA values and therefore can not generate Verilog. But I think this should be solved in magma/compile_guard, *not* in the backend.)

Reply:
Okay, there was already an existing test that has compile guard driving an external value, so we should remove that and we can just raise an error when detecting externals

Reply:
Oh I see. Do we use this in customer code? In that case we would want to continue support and just fix the case that's not working.

Reply:
I doubt it now that I look at the existing test, one of them is incomplete (doesn't fully compile). Likely a WIP effort that wasn't completed, we can bring this up next week and see if anyone wants us to clean it up, but for now I'd rather not have it in half way.